frontend/packages/dev-console/src/utils/shared-submit-utils.ts

```diff
@@ -43,8 +43,9 @@ export const createService = (
 
   let ports = imagePorts;
   const buildStrategy = formData.build?.strategy;
-  if (buildStrategy === 'Docker') {
-    const port = { containerPort: _.get(formData, 'docker.containerPort'), protocol: 'TCP' };
+  if (buildStrategy === 'Docker' && unknownTargetPort) {
```

Comment:
Since you haven't removed the check for `unknownTargetPort` so even if there is no `unknownTargetPort`, `defaultUnkownPort` will not be used instead and it beats the purpose of my previous comment.